app/bundles/FormBundle/Form/Type/FieldType.php

```diff
@@ -388,6 +388,22 @@ public function buildForm(FormBuilderInterface $builder, array $options): void
                     ],
                 ]
             );
+
+            $isReadOnlyValue = (!isset($options['data']['isReadOnly'])) ? false : (bool) $options['data']['isReadOnly'];
+            $builder->add(
+                'isReadOnly',
+                YesNoButtonGroupType::class,
+                [
+                    'label' => 'mautic.form.field.form.read_only',
+                    'data'  => $isReadOnlyValue,
+                    'attr'  => [
+                        'class'        => 'read-only-data',
+                        'tooltip'      => 'mautic.form.field.help.auto_fill',
+                        'data-hide-on' => '{"formfield_isAutoFill_0": "checked"}',
+                        'data-show-on' => '{"formfield_isAutoFill_1": "checked"}',
```

Comment:
I think that @andersonjeccel prefers from the UX perspective to use `data-enable-on` and `data-disable-on` instead of hiding the option completely.